### ISSUE TITLE
Canary regression on amp-list: many responses might be newly failing

### ISSUE THREAD

**Issue by dvoytenko** [ampproject/amphtml#8814]
The error:
```
Response is undefined
```

Most likely cause is absence of AMP CORS headers in responses (CORS headers are fine). Are AMP CORS a new requirement? Are they absolutely necessary for CURLS case?

Some common pages:
https://allrecipes-com.cdn.ampproject.org/c/s/allrecipes.com/recipe/219066/coconut-cheese-flan-flan-de-coco-y-queso/amp/
https://amp-tubiba-com-tr.cdn.ampproject.org/c/s/amp.tubiba.com.tr/ozer-kardesler-fatih

Old masking trace (fixed by #8818):
```
TypeError: Cannot read property 'trim' of undefined
    at ra.f.assert (https://cdn.ampproject.org/rtv/001492217922432/v0.js:7:223)
    at https://cdn.ampproject.org/rtv/001492217922432/v0/amp-list-0.1.js:7:347
```

**Comment by dvoytenko** [comment]
It looks like at least some examples indeed have broken JSON returned on way or another. So it might be mostly publisher issue. However, the splash of recent errors seems suspicious.

**Comment by dvoytenko** [comment]
Confirmed: the issue is fixed by #8818
